lib/instrumentation/modules/@aws-sdk/client-s3.js

```diff
@@ -135,10 +135,20 @@ function s3MiddlewareFactory (client, agent) {
           // use an Access Point bucket ARN for a region *other* than the
           // one for which the client is configured. Therefore, we attempt
           // to get the bucket region from the ARN first.
-          const useArnRegion = await client.config.useArnRegion()
-          const region = useArnRegion && args.input.Bucket.startsWith('arn:')
-            ? regionFromS3Arn(args.input.Bucket)
-            : await client.config.region()
+          const config = client.config
+          let useArnRegion
+          if (typeof config.useArnRegion === 'boolean') {
+            useArnRegion = config.useArnRegion
+          } else {
+            useArnRegion = await config.useArnRegion()
+          }
```

Comment:
Have you found this is necessary? I really hope not.
Did that change in some older `@aws-sdk/client-s3` version perhaps?